mail/spring-cloud-starter-stream-source-mail/src/main/java/org/springframework/cloud/stream/app/mail/source/MailSourceConfiguration.java

```diff
@@ -0,0 +1,157 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.mail.source;
+
+import java.util.Properties;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.app.trigger.TriggerConfiguration;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.SourcePollingChannelAdapterSpec;
+import org.springframework.integration.dsl.mail.Mail;
+import org.springframework.integration.dsl.mail.MailInboundChannelAdapterSpec;
+import org.springframework.integration.dsl.support.Consumer;
+import org.springframework.integration.dsl.support.Transformers;
+import org.springframework.integration.scheduling.PollerMetadata;
+
+/**
+ * A source module that listens for mail and emits the content as a message payload.
+ *
+ * @author Amol
+ */
+@EnableBinding(Source.class)
+@EnableConfigurationProperties({ MailSourceProperties.class })
+@Import({ TriggerConfiguration.class })
+public class MailSourceConfiguration {
+
+	@Autowired
+	@Qualifier("defaultPoller")
+	PollerMetadata defaultPoller;
+
+	@Autowired
+	@Bindings(MailSourceConfiguration.class)
+	Source source;
+
+	@Autowired
+	MailSourceProperties properties;
+
+	@Bean
+	public IntegrationFlow mailInboundFlow() {
+
+		IntegrationFlowBuilder flowBuilder;
+
+		flowBuilder = getFlowBuilder();
+
+		return flowBuilder.transform(Transformers.fromMail()).channel(source.output())
+				.get();
+	}
+
+	/**
+	 * Method to build Integration Flow for Mail. Suppress Warnings for
+	 * MailInboundChannelAdapterSpec.
+	 * @return Integration Flow object for Mail Source
+	 */
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	private IntegrationFlowBuilder getFlowBuilder() {
+		Properties javaMailProperties = this.properties.getJavaMailProperties();
+		String mailURL = this.properties.getProtocol() + "://"
+				+ this.properties.getUsername() + ":" + this.properties.getPassword()
+				+ "@" + this.properties.getHost() + ":" + this.properties.getPort() + "/"
+				+ this.properties.getFolder();
```

Comment:
Yeah, yeah. I see that XD configuration, too.
And I can say that it is has the same vulnerabilities, when some of options aren't required and you miss it during definition.

I still insist that we should use only single `url` property with the sample in docs for proper syntax.
Although it may depend of the vendor like it is with JDBC. And I'm sure you know that we configure there only single URL:
```
spring.datasource.url= # JDBC url of the database.
```
No, any `host`/`port`, just one long url to meet the target server requirements.
Although I may agree that we can extract `login`/`password` as they can be pushed to the `javaMailProperties` and will be used properly from the target Mail Session:
http://connector.sourceforge.net/doc-files/Properties.html